transcom/mymove
# INT B-19903 unlocking a move

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19903)

## Summary

When setting up the functionality of an office user locking a move when they navigate into it, this sets the stage for when an office user needs to unlock that move. Currently there are two ways an office user can unlock a move:
1. Navigate to their respective queue (TOO -> `MovesQueue`, TIO -> `PaymentRequestQueue`, SC -> `ServicesCounselingQueue`)
2. Logout

This PR does the following:
- adds two service objects, `UnlockMove` and `CheckForLockedMovesAndUnlock` to be run for each scenario of an office user unlocking a move
- adds in logic checks to call `UnlockMove` when checks return `true` when loading up the queue list
- calls `checkForLockedMovesAndUnlock` when an office user logs out
- adds relevant tests

### How to test

This is a little tricky to test, so it is best to use two browsers that don't share cookies (I use one Chrome and one Safari - you could also do an incognito browser and one not)
1. Prior to firing up `server` & `client`, in your `envrc` change `FEATURE_FLAG_MOVE_LOCK=true`
2. Now run `make server_run` and `make client_run`
3. (OfficeUser1) Log in as an office user for each role and navigate into a move, but don't log out
4. (OfficeUser2) In your other browser log in as a different user and you should see these moves have a lock icon next to them in the queue
5. (OfficeUser1) Navigate back to your queue
6. (OfficeUser2) Refresh your screen - the lock icons should be gone
7. (OfficeUser1) Navigate back into a move
8. (OfficeUser2) Refresh your screen - lock icon should be shown next to the move
9. (OfficeUser1) Logout
10. (OfficeUser2) Refresh your screen - lock icon should be gone
